Remora.Results/Result.cs

```diff
@@ -32,6 +32,17 @@ namespace Remora.Results;
 [PublicAPI]
 public readonly struct Result : IResult
 {
+    /// <summary>
+    /// Gets a successful result.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// This property returns a result whose <see cref="Error"/> property is set to null, therefore <see cref="IsSuccess"/> returns true.
+    /// The <see cref="Inner"/> property is also set to null.
+    /// </para>
+    /// </remarks>
+    public static IResult SuccessfulResult { get; } = FromSuccess();
```

Comment:
I'd prefer the use of the full type here instead of the interface, since this will preclude the most common usage of this property - returning when the return type is known and not the interface. On the flip side, I'd support a separate property for exposing a boxed singleton - `BoxedSuccess`, perhaps?

Reply:
Property changed to use the full type and to be a computed one based on `FromSuccess()` (or `new(...)` for the typed result).

Added a `BoxedSucess` property to both result types.